.travis.yml

```diff
@@ -1,31 +1,42 @@
-sudo: required
-language: c
+---
+language: haskell
+cabal: "3.4"
 
 os: linux
 dist: xenial
 
-env:
- - GHCVER=8.0.2
- - GHCVER=8.2.2
- - GHCVER=8.4.4
- - GHCVER=8.6.4
+# Caching so the next build will be fast too.
+cache:
+  directories:
+    - $HOME/.stack
+    - $TRAVIS_BUILD_DIR/.stack-work
 
 before_install:
- - sudo add-apt-repository -y ppa:hvr/ghc
- - sudo apt-get update
- - sudo apt-get install cabal-install-1.24 ghc-$GHCVER
- - export PATH=/opt/cabal/bin:/opt/ghc/$GHCVER/bin:$PATH
+  # Download and unpack the stack executable
+  - mkdir -p ~/.local/bin
+  - export PATH=$HOME/.local/bin:$PATH
+  - >
+    travis_retry curl -L https://get.haskellstack.org/stable/linux-x86_64.tar.gz
+    | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
+  # Configure stack to use the system GHC installation
+  - stack config set system-ghc --global true
 
 install:
- - cabal-1.24 update
- - cabal-1.24 install --only-dependencies --enable-tests
-
-script:
- - cabal-1.24 configure --enable-tests
- - cabal-1.24 build
- - dist/build/test/test
- - dist/build/test-io/test-io
- - cabal-1.24 haddock
- - cabal-1.24 sdist
+  - cabal install --only-dependencies
```

Comment:
That only builds the dependencies. It does not build `http-proxy` nor does it run the tests.

Reply:
This is the `install` stage, which is [by default](https://docs.travis-ci.com/user/languages/haskell/#dependency-management) `cabal install --only-dependencies --enable-tests`, but `--enable-tests` [fails](https://travis-ci.community/t/cabal-3-4-not-supported/10453) on Cabal 3.4, so I removed it.
Without defining `script`, the default `cabal build && cabal test` is applied: https://travis-ci.org/github/erikd/http-proxy/jobs/741562080#L936